fs/fd.h

```diff
@@ -55,6 +55,14 @@ struct fd {
             // locked by fd->lock
             struct list unix_scm;
         } socket;
+
+        struct {
+            uint64_t generation;
+            void* wbuffer;
+            size_t wbuffer_off;
+            size_t wbuffer_len;
+            size_t wbuffer_size;
```

Comment:
What's the difference between "len" and "size?

Reply:
see https://github.com/tbodt/ish/pull/447#issuecomment-516057341